### PR TITLE
Upgrade to api-extractor 7

### PR DESCRIPTION
**Pull request by natalieethell**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [X] Include a change request file using `$ npm run change`

#### Description of changes

Upgrade OUFR to api-extractor 7 (AE7)

Main changes:
* api.ts files will now be api.md (Markdown) files
* AE7 supports way, way more of our types. We won't see any more of the "// WARNING: Unsupported export: ..." comments

#### Focus areas to test

Ensure that when making a change to the API, 'rush build' or 'npm run build' in the package directory catches it. Additionally, 'npm run update-api' should correctly update the api.md file.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8459)

### CODE REVIEW

File: packages/date-time/etc/date-time.api.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## API Review File for "@uifabric/date-time"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+
+// (No @packageDocumentation comment for this package)
```
Comment (ecraig12345) [inline packages/date-time/etc/date-time.api.md]: Can we add some basic `@packageDocumentation`?

File: packages/merge-styles/etc/merge-styles.api.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
+## API Review File for "@uifabric/merge-styles"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+// @public
+declare function concatStyleSets<TStyleSet extends IStyleSet<TStyleSet>>(styleSet: TStyleSet | false | null | undefined): IConcatenatedStyleSet<TStyleSet>;
```
Comment (ecraig12345) [inline packages/merge-styles/etc/merge-styles.api.md]: This is cool!  I think it's an improvement over the .api.ts files (also doesn't make VS Code display tons of errors).

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.base.tsx [hunk 3]
```diff
@@ -884,10 +884,9 @@ export class DetailsListBase extends BaseComponent<IDetailsListProps, IDetailsLi
    * which will measure the column cells of all the active rows and resize the
    * column to the max cell width.
    *
-   * @private
-   * @param {IColumn} column (double clicked column definition)
-   * @param {number} columnIndex (double clicked column index)
-   * @todo min width 100 should be changed to const value and should be consistent with the
+   * @param column - (double clicked column definition)
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.base.tsx]: Can you remove the parentheses from this and the next line?

**Comment by KevinTCoughlin** [comment]
I'm curious what the delta is in what we're reporting between AE 6 & 7. The change in file type may make that diff more difficult than its worth to grab.

**Comment by KevinTCoughlin** [comment]
@natalieethell there are a few references to `.api.ts` and similar that we may want to update in addition to the Microsoft Github Bot label feature.

https://github.com/OfficeDev/office-ui-fabric-react/blob/ae66c05a20ddf23c37fd92a1d3528815e15cc45c/.prettierignore#L5

https://github.com/OfficeDev/office-ui-fabric-react/blob/ae66c05a20ddf23c37fd92a1d3528815e15cc45c/scripts/lint-staged/tslint.js#L58

**Comment by natalieethell** [comment]
> I'm curious what the delta is in what we're reporting between AE 6 & 7. The change in file type may make that diff more difficult than its worth to grab.

Here's a snapshot of the difference between the merge-styles API files.
Screen Shot 2019-03-25 at 4 42 14 PM

**Comment by natalieethell** [comment]
Not sure what the build is complaining about since I have the updated merge-styles.api.md file committed - I'll look into this

**Pull request merged**